### PR TITLE
What

### PR DESCRIPTION
**Pull request by RPrudden**

### CODE REVIEW

File: dre/what_decision.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+        """
+        self.whatFilter = whatFilter
+        self.timeSlot = timeSlot
+        self.location = location
+        self.cache = cache
+
+        self.possibleActivities = []
+
+    def generatePossibleActivities(self, actionTimeRes):
```
Comment (niallrobinson) [inline dre/what_decision.py]: methods and variable names should be underscore separated. Only classes are camel case

Comment (niallrobinson) [inline dre/what_decision.py]: maybe actionTimeRes should have a default value of the same time res as the forecast

File: dre/what_decision.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+                generating Actions within an Activity.
+
+        """
+        for activity in self.whatFilter:
+            if activity.duration <= (self.timeSlot.maxTime - self.timeSlot.minTime):
+                possibility = []
+                t = self.timeSlot.minTime
+                while t < self.timeSlot.maxTime:
+                    templateAction = WhatActionBuilder(self.location, self.cache)
+                    possibility.append(templateAction.build(activity.score, activity.conditions, t))
```
Comment (niallrobinson) [inline dre/what_decision.py]: could the scoring method be in the `WhatActionBuilder.__init__`

File: dre/what_decision.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+            * actionTimeRes (datetime.timedelta): Defines the resolution for
+                generating Actions within an Activity.
+
+        """
+        for activity in self.whatFilter:
+            if activity.duration <= (self.timeSlot.maxTime - self.timeSlot.minTime):
+                possibility = []
+                t = self.timeSlot.minTime
+                while t < self.timeSlot.maxTime:
+                    templateAction = WhatActionBuilder(self.location, self.cache)
```
Comment (niallrobinson) [inline dre/what_decision.py]: this should be outside the while loop I think

File: dre/what_decision.py [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+        Args:
+            * actionTimeRes (datetime.timedelta): Defines the resolution for
+                generating Actions within an Activity.
+
+        """
+        for activity in self.whatFilter:
+            if activity.duration <= (self.timeSlot.maxTime - self.timeSlot.minTime):
+                possibility = []
+                t = self.timeSlot.minTime
+                while t < self.timeSlot.maxTime:
```
Comment (niallrobinson) [inline dre/what_decision.py]: what if a run takes 10 minutes and you've got a timeSlot of two hours? Shouldn't this be:
```
if activity.duration <= (self.timeSlot.maxTime - self.timeSlot.minTime):
    t = self.timeSlot.minTime
    while t < (self.timeSlot.minTime+activity.duration):
```
or something

File: dre/what_decision.py [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+
+    def generatePossibleActivities(self, actionTimeRes):
+        """
+        Populates possibleActivites with valid Activities, ordered by quality.
+        Args:
+            * actionTimeRes (datetime.timedelta): Defines the resolution for
+                generating Actions within an Activity.
+
+        """
+        for activity in self.whatFilter:
```
Comment (niallrobinson) [inline dre/what_decision.py]: can we change `activity` to `this_activity`. It gets confusing later on when we've got `Activity` as well

File: dre/what_decision.py [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+    def __init__(self, whatFilter, timeSlot, location, cache):
+        """
+        Args:
+            * whatFilter (list): List of WhatActivity objects
+            * timeSlot (TimeSlot): Start and end of possible time window
+            * location (Loc): Location of action
+            * cache (forecastCache.cache): Forecast cache
+
+        """
+        self.whatFilter = whatFilter
```
Comment (niallrobinson) [inline dre/what_decision.py]: I know this is to be comparable to `timeFilter` but maybe we should change this to `possible_activities` or something

**Comment by RPrudden** [comment]
Done functional changes - leaving naming conventions for a later PR.

**Pull request merged**